### PR TITLE
Fetchall

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

File: odo/backends/sql.py [hunk 1]
```diff
@@ -321,7 +320,7 @@ def select_to_base(sel, dshape=None, **kwargs):
     engine = sel.bind  # TODO: get engine from select
 
     with engine.connect() as conn:
-        result = conn.execute(sel)
+        result = conn.execute(sel).fetchall()
```
Comment (mrocklin) [inline odo/backends/sql.py]: Does this affect laziness at all?

Comment (cpcloud) [inline odo/backends/sql.py]: Hm. Laziness *does* have the benefit of not blowing up memory, but without `fetchall` sqlalchemy can and often does hit the server with every call to `fetchone`. This of course depends on the dialect. I like laziness but hitting the server every single time for, say, a 1000 row result set is crazy. A happy medium might be to do something along the lines of:

```python
result = conn.execute(sel)
chunksize = 1000

def execute(conn, sel, chunksize=1000):
    result = conn.execute(sel)
    while True:
        try:
            yield result.fetchmany(size=chunksize)
        except StopIteration:
            return

for row in concat(execute(conn, sel)):
    yield row
```

Comment (mrocklin) [inline odo/backends/sql.py]: Batching sounds good.  I'm surprised that sqlalchemy doesn't do this
already.

**Comment by cpcloud** [comment]
closing in favor of a new pr that will have batching

**Pull request closed without merging**